src/imitation/rewards/reward_nets.py

```diff
@@ -533,6 +542,52 @@ def forward(
         assert final_rew.shape == state.shape[:1]
         return final_rew
 
+    def predict_th(
```

Comment:
This looks copy-pasted from `RewardNet`, can we just do `predict_th = RewardNet.predict_th`? I think this works because the `self` will still be different between them. Similar for the following methods.

Reply:
This does not work since when doing this it does not automatically curry the method, so you would have to manually pass in self.

Reply:
Can you write something like `def predict_th(self, s, a, s_) = RewardNet.predict_th(self, s, a, s_)`?

Reply:
I just made a commit changing it along the lines I suggested and the tests still pass. Feel free to revert if me/the test cases are missing something, but could you give an example of where it breaks?

Reply:
Okay I think this works. I was adding it as an attribute in the `__init__` method. Which causes errors.

Reply:
I was working on a slightly different solution. That I think might be a bit more general.

Add two subclasses of `RewardNetWrapper`: `PredictProcessedWrapper` and `ForwardWrapper`. `PredictProcessedWrapper` requires you to override `predict_processed` and defaults to calling the base RewardNet for everything. `ForwardWrapper` requires you to override `forward` cannot be placed on top of a `PredictProcessedWrapper` wrapper and uses pythons default method resolution to call the super class.

Then `ShapedRewardNet` would be a `ForwardWrapper` and all of our other current wrappers would be subclasses of `PredictProcessedWrapper`.

Reply:
FWIW I quite like this idea.